### ISSUE TITLE
Parsing error while creating clone using fully qualified name [Dialect : Snowflake]

### ISSUE THREAD

**Issue by arumugam-pounraj** [sqlfluff/sqlfluff#3413]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Created a SQL statement for cloning the schema in snowflake. I have used fully qualified name in the sql statement 
**[i.e DatabaseName.SchemaName]** and tried executing `sqlfluff lint`. It fails with Parsing error

**SQL Statement**
`CREATE SCHEMA TEST_DB.TEST_CLONED_SCHEMA CLONE TEST_DB.TEST_SCHEMA;`

**Error**

`sqlfluff lint clone_test.sql`

![SQLFluff_Result](https://user-images.githubusercontent.com/79456473/171132830-9212065d-a67f-4b2d-acd8-1272faf923e4.JPG)

But if the same statement is executed without fully qualified name then it doesn't show any parsing error

### Expected Behaviour

The parsing error shouldn't have occurred even while using the fully qualified name **[i.e DatabaseName.SchemaName]**

### Observed Behaviour

Getting Parsing error when using the fully qualified name. **[i.e DatabaseName.SchemaName]**
So far observed this issue only while using the CLONE command in snowflake

### How to reproduce

Create  **clone_test.sql**  file with below contents
`CREATE SCHEMA TEST_DB.TEST_CLONED_SCHEMA CLONE TEST_DB.TEST_SCHEMA;`

Execute the below lint command
` sqlfluff lint  clone_test.sql`

### Dialect

Snowflake

### Version

` sqlfluff version`
0.13.2

### Configuration

```
[sqlfluff]
dialect = snowflake

[sqlfluff:rules:L014]
# Unquoted identifiers
extended_capitalisation_policy = consistent

[sqlfluff:rules:L062]
# Comma separated list of blocked words that should not be used
blocked_words = USE

[sqlfluff:rules:L016]
# Comma separated list of blocked words that should not be used
max_line_length = 400
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by mdahlman** [comment]
@arumugam-pounraj , your bug was nicely explained. It made it easy to find the issue quickly.

**Comment by tunetheweb** [comment]
Now you’re just making me look bad, since I argued it on slack! 😁

**Comment by mdahlman** [comment]
Ha! I actually laughed out loud about that comment. 
I was curious about the exact details of why it parsed with non-fully qualified names. Imagine my surprise when `create_clone_statement` showed up in the parse structure. So the fix became unexpectedly easy.
